src/courses/mappers/18.md

```diff
@@ -0,0 +1,271 @@
+---
+order: 18
+title: Appendix - SAF CLI Integration
+author: Charles Hu
+---
+
+## SAF CLI Integration
+
+The following is a supplemental lesson on how to integrate your developed mapper with the SAF CLI. Integration with the SAF CLI will allow your mapper to be used in a command line environment, independent of Heimdall2.
+
+If you have not yet created a mapper, please follow the primary course and do so. Once completed, you will be able to continue with this lesson.
+
+### Set Up
+
+First, we need set up the necessary files to begin integrating your mapper with the SAF CLI.
+
+1. Create a development branch against the [SAF CLI repository](https://github.com/mitre/saf) and create a draft pull request for your new branch.
+
+2. In the `package.json` file, update the versions of `@mitre/hdf-converters` and `@mitre/heimdall-lite` to the [latest release of Heimdall2](https://github.com/mitre/heimdall2/releases).
+
+3. In the `src/commands/convert` directory, create a blank TypeScript file. It should be named:
+```
+{YOUR-EXPORT-NAME-HERE}2hdf.ts
+```
+
+4. In the `test/sample_data` directory, create a directory named `{YOUR-EXPORT-NAME-HERE}`. Underneath it, create a directory named `sample_input_report`. The file structure should now look like this:
+```
++-- sample_data
+|   +-- {YOUR-EXPORT-NAME-HERE}
+|   |   +-- sample_input_report
+```
+
+5. Place your sample export under the `sample_input_report` directory. Your sample export should be genericized to avoid any leaking of sensitive information. Under the `{YOUR-EXPORT-NAME-HERE}` directory, place your output OHDF files generated during the original testing phase of your mapper development. The file structure should now look like this:
+```
++-- sample_data
+|   +-- {YOUR-EXPORT-NAME-HERE}
+|   |   +-- sample_input_report
+|   |   |   +-- {YOUR-SAMPLE-EXPORT}
+|   |   +-- {YOUR-EXPORT-NAME-HERE}-hdf.json
+|   |   +-- {YOUR-EXPORT-NAME-HERE}-hdf-withraw.json
+```
+
+6. In the `test/commands/convert` directory, create a blank TypeScript file. It should be named:
+```
+{YOUR-EXPORT-NAME-HERE}2hdf.test.ts
+```
+
+### Integration
+
+With file set up out of the way, we can begin filling out the necessary files to create a `saf convert` command for our mapper.
+
+1. Insert the skeleton convert command file (see below) in the file `{YOUR-EXPORT-NAME-HERE}2hdf.ts` which you created earlier. Replace names (`SKELETON` by default) as necessary.
+
+::: details Skeleton Convert Command File
+```typescript
+import {Command, Flags} from '@oclif/core'
+import fs from 'fs'
+import {SKELETONMapper as Mapper} from '@mitre/hdf-converters'
+import {checkSuffix} from '../../utils/global'
+
+export default class SKELETON2HDF extends Command {
+  static usage = 'convert SKELETON2hdf -i <SKELETON-json> -o <hdf-scan-results-json>'
```

Comment:
might be worthwhile to link the docopts site for help on writing good usage strings: http://docopt.org/

Reply:
Addressed.